http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/HttpAdvancedReactiveIT.java

```diff
@@ -163,14 +198,146 @@ public void vertxHttpClientRedirection() throws InterruptedException, URISyntaxE
+                .body(HelloResource.QUERY_PARAM_TITLE, is("Hello Title"));
+    }
+
+    /**
+     * Test must be enabled with property {@link NinetyNineBottlesOfBeerResource#ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_NAME}
+     * set to {@link NinetyNineBottlesOfBeerResource#ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_VAL}.
+     *
+     * @see NinetyNineBottlesOfBeerResource for more information
+     */
+    @EnabledIfSystemProperty(matches = ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_VAL, named = ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_NAME, disabledReason = TEST_REQUIRES_AT_LEAST_2_8_3)
+    @DisabledOnQuarkusVersion(version = "2\\.[0-9]\\..*", reason = "Fixed in Quarkus 2.10")
+    @DisplayName("JAX-RS RouterFilter and Vert.x Web Routes integration")
+    @Test
+    public void multipleResponseFilter() {
+        // test headers from both filters are present, that is useful content negotiation
+        // scenario -> server side should be able to set multiple VARY response headers
+        // so browser can identify when to serve cached response and when to send request to a server side
+        var headers = app.given().get(HELLO_ENDPOINT).headers().asList();
+        assertHasHeaderWithValue(HttpHeaders.ACCEPT_LANGUAGE, ENGLISH, headers);
+        assertHasHeaderWithValue(HttpHeaders.ACCEPT_LANGUAGE, JAPANESE, headers);
+        assertHasHeaderWithValue(HttpHeaders.ACCEPT_ENCODING, ANY_ENCODING, headers);
+        assertHasHeaderWithValue(HttpHeaders.VARY, ACCEPT_ENCODING, headers);
+        assertHasHeaderWithValue(HttpHeaders.VARY, ACCEPT_LANGUAGE, headers);
+    }
+
+    @DisplayName("Several Resources share same base path test")
+    @Test
+    public void severalResourcesSameBasePath() {
+        // following endpoints are placed in 2 different Resources with the same base path
+        app.given().get(HELLO_ENDPOINT).then().body(CONTENT, is("Hello, World!"));
+        app.given().get(HELLO_ENDPOINT + HelloAllResource.ALL_ENDPOINT_PATH).then().body(CONTENT, is("Hello all, World!"));
+    }
+
+    private void assertHasHeaderWithValue(String headerName, String headerValue, List<Header> headers) {
+        Assertions.assertTrue(
+                headers
+                        .stream()
+                        .filter(h -> h.getName().equals(headerName))
+                        .map(Header::getValue)
+                        .anyMatch(headerValue::equals));
+    }
+
+    /**
+     * Test must be enabled with property {@link NinetyNineBottlesOfBeerResource#ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_NAME}
+     * set to {@link NinetyNineBottlesOfBeerResource#ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_VAL}.
+     *
+     * @see NinetyNineBottlesOfBeerResource for more information
+     */
+    @EnabledIfSystemProperty(matches = ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_VAL, named = ENABLED_ON_QUARKUS_2_8_3_OR_HIGHER_NAME, disabledReason = TEST_REQUIRES_AT_LEAST_2_8_3)
+    @DisabledOnQuarkusVersion(version = DISABLE_IF_NOT_QUARKUS_2_8_3_OR_HIGHER, reason = TEST_REQUIRES_AT_LEAST_2_8_3)
```

Comment:
The issue has been fixed in `2.7.6.Final`, `2.8.3.Final`, and `2.9.0.Final` (+ `main`). Please adjust.

Reply:
Adjusted to 2.7.6, 2.8.3 or higher. I quickly tested it and fix worked.